nullaway/src/main/java/com/uber/nullaway/handlers/StreamNullabilityPropagator.java

```diff
@@ -204,14 +217,62 @@ public void onMatchMethodInvocation(
             ClassTree annonClassBody = ((NewClassTree) argTree).getClassBody();
             // Ensure that this `new B() ...` has a custom class body, otherwise, we skip for now.
             if (annonClassBody != null) {
-              MaplikeMethodRecord methodRecord = streamType.getMaplikeMethodRecord(methodSymbol);
-              handleMapAnonClass(methodRecord, tree, annonClassBody);
+              MapLikeMethodRecord methodRecord = streamType.getMaplikeMethodRecord(methodSymbol);
+              handleMapOrCollectAnonClassBody(
+                  methodRecord,
+                  annonClassBody,
+                  t -> observableCallToInnerMethodOrLambda.put(tree, t));
             }
           } else if (argTree instanceof LambdaExpressionTree) {
             observableCallToInnerMethodOrLambda.put(tree, argTree);
           } else if (argTree instanceof MemberReferenceTree) {
             observableCallToInnerMethodOrLambda.put(tree, argTree);
           }
+        } else {
+          CollectLikeMethodRecord collectlikeMethodRecord =
+              streamType.getCollectlikeMethodRecord(methodSymbol);
+          if (collectlikeMethodRecord != null && methodSymbol.getParameters().length() == 1) {
+            handleCollectCall(tree, collectlikeMethodRecord);
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Handles a call to a collect-like method. If the argument to the method is supported, updates
+   * the {@link #collectCallToInnerMethodsOrLambdas} map appropriately.
+   */
+  private void handleCollectCall(
+      MethodInvocationTree tree, CollectLikeMethodRecord collectlikeMethodRecord) {
+    ExpressionTree argTree = tree.getArguments().get(0);
+    if (argTree instanceof MethodInvocationTree) {
+      MethodInvocationTree collectInvokeArg = (MethodInvocationTree) argTree;
```

Comment:
This one is the `Collectors.toMap(...)` invocation, right?

Reply:
Yup, 8b3f984